test/routes/add-article.test.ts

```diff
@@ -28,4 +28,18 @@ describe('add article', (): void => {
   it('should throw an error if id is already set', async (): Promise<void> => {
     await expect(makeRequest(createArticle('_:1'))).rejects.toThrow(new createHttpError.Forbidden('Article IDs must not be set (\'_:1\' was given)'));
   });
+
+  it('should throw an error if it is not a schema:Article', async (): Promise<void> => {
+    await expect(makeRequest({
+      '@id': '_:1',
+      '@type': 'http://schema.org/NewsArticle',
+      'http://schema.org/name': 'Article _:1',
+    })).rejects.toThrow(new createHttpError.Forbidden('Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)'));
+  });
+
+  it('should throw an error if no scheam:name set', async (): Promise<void> => {
+    await expect(makeRequest({
+      '@type': 'http://schema.org/Article',
+    })).rejects.toThrow(new createHttpError.Forbidden('Article must have at least one http://schema.org/name'));
```

Comment:
there are two scenarios to test that the check is catering for: missing field and a field evaluating to an empty string

Reply:
Have just added these scenarios